rclpy/rclpy/task.py

```diff
@@ -16,22 +16,17 @@
 import inspect
 import sys
 import threading
-from types import CoroutineType
 from typing import (Any, Callable, cast, Coroutine, Dict, Generator, Generic, List,
-                    Optional, Tuple, TYPE_CHECKING, TypeVar, Union)
+                    Optional, overload, Tuple, TYPE_CHECKING, TypeVar, Union)
 import warnings
 import weakref
 
 if TYPE_CHECKING:
-    from typing import TypeAlias
 
     from rclpy.executors import Executor
 
 T = TypeVar('T')
 
-FunctionOrCoroutineFunction: 'TypeAlias' = Union[Callable[..., T],
-                                                 Callable[..., Coroutine[None, None, T]]]
```

Comment:
Just a note: I would generally be weary of simply removing something, but it was added 2 months ago in #1370, so it's fine